collector/transport_node_collector.go

```diff
@@ -52,36 +65,39 @@ func (c *transportNodeCollector) Describe(ch chan<- *prometheus.Desc) {
 
 // Collect implements the prometheus.Collector interface.
 func (c *transportNodeCollector) Collect(ch chan<- prometheus.Metric) {
-	transportNodeMetrics := c.generateTransportNodeMetrics()
+	transportNodes, err := c.transportNodeClient.ListAllTransportNodes()
+	if err != nil {
+		level.Error(c.logger).Log("msg", "Unable to list transport nodes", "err", err)
+		return
+	}
+	edgeClusterMemberships, err := c.generateEdgeClusterMemberships()
+	if err != nil {
+		edgeClusterMemberships = nil
```

Comment:
why not returning nil from `generateEdgeClusterMembership` instead?

Reply:
`generateEdgeClusterMemberships` does return `nil` when failed
this is just to make sure, as from `Collect` side there's no guarantee that `generateEdgeClusterMemberships` will return `nil`

Reply:
ok make sense